protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,80 @@
+# Network wide limits
+
+This spec describes network wide limits aimed at keeping the overall system performant and responsive.
+Vega has been designed with low-latency in mind so that the responsiveness of the network doesn't unduly impact the trading experience. Furthermore, the current implementation relies on an interplay between the execution layer (`core`) and the data consumption layer (`data node`). For these reasons limits allowing to exercise a degree of control over the computational and data generation loads need to be put in place. These limits need to be adjustable so that each network/shard can be setup for maximum accessibility (hardware constraints permitting) while preserving the desirable user experience and system stability, and mitigating the possibility of a malicious actor trying to deliberately disrupt the system by flooding it with superfluous requests.
+
+## Number of assets
+
+`limits.assets.max`
+
+TODO: Do we really care? Each asset is added via a governance vote, so it doesn't seem likely we'll get a ton of unnecessary ones (plus we'd then have to implement a process of removing an asset with a vote if the limit was already reached, but there was some new asset that community perceived as very important). Also, does an additional asset itself really increase the load that much?
+
+## Number of markets
+
+Introduce a new network parameter `limits.markets.max` controlling the maximum number of markets allowed within a network.
+
+Each market in a `Proposed` [state](./0043-MKTL-market_lifecycle.md) should increment the **total market count** by 1. If a market reaches a `Rejected`, `Cancelled`, `Closed` or `Settled` state the total market count should be decremented by 1.
+As soon as the total market count reaches the value specified by `limits.markets.max` parmeter no other further market proposals should be accepted. If the total market count drops below `limits.markets.max` the network should start accepting market proposals again.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
```

Comment:
```suggestion
If the network parameter gets increased via a governance vote no further actions are needed.
```

Reply:
This is just a line drop for readability.